src/metabase/api/activity.clj

```diff
@@ -189,11 +190,78 @@
 (def ^:private views-limit 8)
 (def ^:private card-runs-limit 8)
 
-#_{:clj-kondo/ignore [:deprecated-var]}
-(api/defendpoint-schema GET "/recent_views"
-  "Get the list of 5 things the current user has been viewing most recently."
+(defn- bookmarks
+  [user-id]
+  (let [as-null (when (= (mdb.connection/db-type) :postgres) (h2x/->integer nil))]
+    {:select    [[:type :model] [:item_id :model_id]]
+     :from [[{:union-all [{:select [:card_id
+                                    [as-null :dashboard_id]
+                                    [as-null :collection_id]
+                                    [:card_id :item_id]
+                                    [[:inline "card"] :type]
```

Comment:
I think we're supposed to use `h2x/literal` for hard-coded strings in honeySQL queries

Reply:
I changed that, but to be honest I'm not sure what ends up being different.